dev/tasks/python-wheels/github.windows.yml

```diff
@@ -68,11 +68,12 @@ jobs:
           @rem We can remove this workaround once we find a way to use
           @rem pulled caches when build an image.
           echo on
-          archery docker pull --no-ignore-pull-failures %TEST_IMAGE_PREFIX%-wheel-windows-vs2022
-          if errorlevel 1 (
-            archery docker build --no-pull %TEST_IMAGE_PREFIX%-wheel-windows-vs2022 || exit /B 1
-          )
-          archery docker run --no-build -e SETUPTOOLS_SCM_PRETEND_VERSION={{ arrow.no_rc_version }} %TEST_IMAGE_PREFIX%-wheel-windows-vs2022
+          @rem archery docker pull --no-ignore-pull-failures %TEST_IMAGE_PREFIX%-wheel-windows-vs2022
+          @rem if errorlevel 1 (
+          @rem   archery docker build --no-pull %TEST_IMAGE_PREFIX%-wheel-windows-vs2022 || exit /B 1
+          @rem )
+          @rem archery docker run --no-build -e SETUPTOOLS_SCM_PRETEND_VERSION={{ arrow.no_rc_version }} %TEST_IMAGE_PREFIX%-wheel-windows-vs2022
+          archery docker run -e SETUPTOOLS_SCM_PRETEND_VERSION={{ arrow.no_rc_version }} python-wheel-windows-vs2022
```

Comment:
It looks like this is what "fixes" the windows crossbow jobs. I don't fully understand what is going on here, but I came across this in the comment directly preceding this as a solution.

It seems like the windows crossbow job had a pre-existing issue with the sequence of building an image then executing it through archery (?). I am not sure if this PR needs to solve this or if its just a caching thing that is expected to go away once in main

Reply:
I think you want to bump the image tag instead:
https://github.com/apache/arrow/blob/d2ddee62329eb711572b4d71d6380673d7f7edd1/.env#L95-L100

Reply:
No, you have to do it in this PR. This will ensure it doesn't use any outdated cached image.

Reply:
Looks like that did the trick - thanks!